src/showingpreviously/cinemas/centre_for_the_moving_image.py

```
@@ -0,0 +1,134 @@
+    name = showing_item.find('span', class_='field--name-title').text
+    more_info_link = showing_item.find('a', class_='btn-more-info')['href']
+    release_year, attributes = get_film_page_details(cinema, more_info_link)
+    showing_times = showing_item.find('div', class_='screening-times-btns')
+    film = Film(name=name, year=release_year)
+    for showing_time in showing_times.find_all('a', class_='btn-times'):
+        showing_time_str = showing_time.text.strip().split('\n')[0].strip()
+        showing_datetime = datetime.datetime.strptime(f'{showing_date_str} {showing_time_str}', '%Y-%m-%d %H:%M')
+        showing_screen = Screen(name=showing_time.find('span', class_='screen').text.strip())
+        showing_type_attributes = get_showing_type_attributes(name, showing_date_str, showing_time)
+        showing = Showing(
+            film = film,
+            time = showing_datetime,
+            chain = CHAIN,
+            cinema = cinema['cinema'],
+            screen = showing_screen,
+            json_attributes = {**attributes, **showing_type_attributes}
+        )
+        showing_list.append(showing)
+    return showing_list
+
+
+def get_showing_type_attributes(film_name: str, showing_date_str: str, showing: BeautifulSoup) -> dict[str, any]:
+    showing_type_list = showing.find_all('span', class_='showing-type')
+    attributes = {}
+    for showing_type in showing_type_list:
+        showing_type_attribute = [attr for attr in showing_type['class'] if attr != 'showing-type']
+        if len(showing_type_attribute) != 1:
+            raise CinemaArchiverException('Expected one showing type attribute for {film_name} on {showing_date_str}: {str(showing_type_attribute)}')
+        showing_type_attribute = showing_type_attribute[0]
+        if showing_type_attribute not in KNOWN_SHOWING_TYPE_ATTRIBUTES:
+            raise CinemaArchiverException('Unknown showing type attribute for {film_name} on {showing_date_str}: {showing_type_attribute}')
```

Do we want to raise an exception here? Or log it (somehow) and continue processing?

---

We don't have a proper logging framework (yet) and I want to ensure that any new attributes are caught as they are introduced. Raising an exception seems to be the best way to do this, at least for now.